core/src/main/scala/org/broadinstitute/dsde/rawls/user/UserService.scala

```diff
@@ -851,13 +852,35 @@ class UserService(
     billingAccount: Option[RawlsBillingAccountName]
   ): Future[Option[RawlsBillingProjectResponse]] = for {
     project <- updateBillingAccountInDatabase(projectName, billingAccount)
+    _ = project.map(updateBillingAccountInBillingProfile(_, billingAccount))
     projectRoles <- samDAO
       .listUserRolesForResource(SamResourceTypeNames.billingProject, projectName.value, ctx)
       .map(resourceRoles => samRolesToProjectRoles(resourceRoles))
   } yield project.flatMap { p =>
     if (projectRoles.nonEmpty) Some(RawlsBillingProjectResponse(projectRoles, p)) else None
   }
 
+  private def updateBillingAccountInBillingProfile(project: RawlsBillingProject,
+                                                   billingAccount: Option[RawlsBillingAccountName]
+  ): Unit =
+    try
+      project.billingProfileId.map { billingProfileId =>
+        billingAccount match {
+          case Some(newBillingAccount) =>
+            billingProfileManagerDAO.updateBillingProfile(UUID.fromString(billingProfileId), newBillingAccount, ctx)
+          case None =>
+            billingProfileManagerDAO.removeBillingAccountFromBillingProfile(UUID.fromString(billingProfileId), ctx)
+        }
+      }
+    catch {
+      // Until BPM is the system of record for Terra billing information, Rawls will not throw an exception if BPM fails to update
+      case e: ApiException =>
+        logger.warn(
```

Comment:
Can this go to sentry too? I think even if we're not the SOR we want to know about it and a log entry is very likely to get missed.

Reply:
Good call, I'll make sure they go to sentry

Reply:
In general or just in this case?

In this case, we want to emit an event to Sentry when a billing profile fails to update so we have a sense for how often that's happening and will know when a billing project and its billing profile are out of sync.

Reply:
What I am asking is what controls whether or not an error goes to Sentry-- does the exception need to be rethrown with specific error codes, or we need to call logger.error, or a different method call…?

Reply:
Oh I see, 5xx errors from Rawls will automatically be sent to Sentry, and I think error-level log messages as well? You can also manually send events to Sentry which is what I'm planning on doing here